src/components/kubectl/port-forward.ts

```diff
@@ -176,31 +208,33 @@ async function promptForPort(kubectl?: Kubectl, podName?: string, namespace?: st
     if (!portString) {
         return [];
     }
-    return buildPortMapping(portString);
+    return buildPortMapping(portString, extractedPorts);
 }
 
 /**
  * Validates the user supplied port mapping(s)
  * @param portMapping The portMapping string captured from an input field
+ * @param validPorts List of valid named ports
  * @returns A ValidationResult object describing the first error found.
  */
-function validatePortMapping(portMapping: string): ValidationResult | undefined {
+function validatePortMapping(portMapping: string, validPorts: ExtractedPort[] = []): ValidationResult | undefined {
     const portPairs = portMapping.split(' ');
-    const validationResults: ValidationResult[] = portPairs.map(validatePortPair);
+    const validationResults: ValidationResult[] = portPairs.map(validatePortPair.bind(null, validPorts));
```

Comment:
Rather than using `.bind` could we use a lambda here?
```suggestion
    const validationResults: ValidationResult[] = portPairs.map((pair) => validatePortPair(validPorts, pair));
```

It may be personal taste but I find this easier to read than `bind` with its irksome pseudo-`this` argument... *grin*

Reply:
This might allow you to remove the `: ValidationResult[]` explicit type annotation too.